location/views.py

```diff
@@ -188,53 +188,47 @@ def sort_by_choices(self) -> list[tuple[str, str]]:
             ('change_timestamp', 'Last edit',),
         ]
 
+    def create_queryset_by_queries(self, model_class: Type[models.Model], queries: Iterable[Q]):
+        queryset = model_class.objects.all()
+        annotate = {'sent': Count('works',
+                                  filter=Q(cofkworklocationmap__relationship_type=REL_TYPE_WAS_SENT_FROM)),
+                    'recd': Count('works',
+                                  filter=Q(cofkworklocationmap__relationship_type=REL_TYPE_WAS_SENT_TO)),
+                    }
+        annotate['all_works'] = annotate['sent'] + annotate['recd']
+
+        queryset = queryset.annotate(**annotate)
+
+        if queries:
+            queryset = queryset.filter(query_utils.all_queries_match(queries))
+
+        if sort_by := self.get_sort_by():
+            queryset = queryset.order_by(sort_by)
+
+        return queryset
```

Comment:
suggest reuse super method 
```suggestion
        queryset = super().create_queryset_by_queries(model_class, queries)
        annotate = {'sent': Count('works',
                                  filter=Q(cofkworklocationmap__relationship_type=REL_TYPE_WAS_SENT_FROM)),
                    'recd': Count('works',
                                  filter=Q(cofkworklocationmap__relationship_type=REL_TYPE_WAS_SENT_TO)),
                    }
        annotate['all_works'] = annotate['sent'] + annotate['recd']
        queryset = queryset.annotate(**annotate)
        return queryset
```

Reply:
but what if the query is supposed to be sorted by sent, recd or all_works?

Reply:
not sure will sorted sent work, if not working, we can extract.
```
        if queries:
            queryset = queryset.filter(query_utils.all_queries_match(queries))
        if sort_by := self.get_sort_by():
            queryset = queryset.order_by(sort_by)
```